src/winkerberos.c

```diff
@@ -475,7 +633,19 @@ sspi_client_step(PyObject* self, PyObject* args) {
         return NULL;
     }
 
-    result = auth_sspi_client_step(state, challenge);
+    if (pychan_bindings != NULL) {
+        if (!PyCObject_Check(pychan_bindings)) {
+            PyErr_SetString(PyExc_TypeError, "Expected a SecPkgContext_Bindings object");
+            return NULL;
+        }
+        sec_pkg_context_bindings = (SecPkgContext_Bindings *)PyCObject_AsVoidPtr(pychan_bindings);
+        if (sec_pkg_context_bindings == NULL) {
+            PyErr_SetString(PyExc_RuntimeError, "NULL result returned when getting a SecPkgContext_Bindings struct from a PyCObject");
```

Comment:
If PyCObject_AsVoidPtr (or the equivalent PyCapsule function) fails an exception will automatically be set. You shouldn't raise RuntimeError here. Just return NULL.